Documentation/gettingstarted/local-redirect-policy.rst

```diff
@@ -301,5 +305,67 @@ configuring the CiliumLocalRedirectPolicy.
 changes to be made, delete the existing policy, and re-create a new one.
 
+Real-world Use Cases
+====================
+Local redirect policy allows for some real-world applications to run with Cilium.
+
+Node-local DNS cache
+--------------------
+`DNS node-cache <https://github.com/kubernetes/dns>`_ listens on a static IP to intercept
+traffic from application pods to the cluster's DNS service VIP.
+
+One way to steer traffic to the local DNS node-cache with local redirect policy is to put
+the node-cache in its own pod namespace and let it listen on its own IP.
+Here are a few steps to create such a setup with a Cilium cluster (`example yaml <https://github.com/cilium/cilium/blob/master/examples/kubernetes-local-redirect/node-local-dns.yaml>`_):
+
+- Make sure to use a Node-local DNS image with `PR <https://github.com/kubernetes/dns/pull/407>`_.
+  This is to ensure that we have a knob to disable dummy network interface creation/deletion in
+  Node-local DNS when we deploy it in non-host namespace.
+
+- Modify Node-local DNS cache's deployment yaml to pass these additional arguments:
+  "-skipteardown=true", "-setupinterface=false", and "-setupiptables=false".
+
+- Modify Node-local DNS cache's deployment yaml to put it in non-host namespace by setting
+  "hostNetwork: false" for the daemonset and, in the Corefile, bind to "0.0.0.0" instead of
+  the static IP.
+
+- Deploy the modified Node-local DNS yaml and the `local redirect policy
+  <https://github.com/cilium/cilium/blob/master/examples/kubernetes-local-redirect/node-local-dns-lrp.yaml>`_.
+  Note that you may need to modify the LRP a bit to match the cluster DNS service you have.
+
+kiam redirect on EKS
+--------------------
+`kiam <https://github.com/uswitch/kiam>`_ agent runs on each node in an EKS
+cluster, and intercepts requests going to the AWS metadata server to fetch
+security credentials for pods.
+
+- In order to only redirect traffic from pods to the kiam agent, and pass
+  traffic from the kiam agent to the AWS metadata server without any redirection,
+  we need the socket lookup functionality in the datapath. This functionality
+  requires v5.1.16, v5.2.0 or more recent Linux kernel.
+
+- Add a metadata label for the kiam agent daemonset, as specified in the example
+  `kiam yaml file <https://github.com/cilium/cilium/blob/master/examples/kubernetes-local-redirect/kiam.yaml>`_.
+  The same label is used as the ``localEndpointSelector`` label in the `kiam local redirect policy
+  <https://github.com/cilium/cilium/blob/master/examples/kubernetes-local-redirect/kiam-lrp.yaml>`_.
+
+- The ``addressMatcher`` ip address specified in the local redirect policy is the
+  ip address of the AWS metadata server, and port 80 is the default HTTP server port.
+  The port specified in the ``redirectBackend`` configuration in the policy is the
+  port specified in the kiam yaml file, that the kiam agent listens on.
```

Comment:
This paragraph is unlike the others, the others are typically written in the imperative sense to tell the reader what to do. Configure X, Remove Y from the configuration, etc. On the other hand, this paragraph seems to be describing the LRP. Perhaps we should instead put this outside the list of steps for the user, and have the inline LRP example so that the user can read this text and follow along with the LRP?

Reply:
Done